script/github-actions/check-deployability.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable no-console */
+/* eslint-disable camelcase */
+const path = require('path');
+const fetch = require('node-fetch');
+const core = require('@actions/core');
+const { Octokit } = require('@octokit/rest');
+const { runCommandSync, sleep } = require('../utils');
+
+const BUCKETS = require('../../src/site/constants/buckets');
+const ENVIRONMENTS = require('../../src/site/constants/environments');
+
+const { GITHUB_TOKEN: auth, GITHUB_REPOSITORY, GITHUB_SHA } = process.env;
+const [owner, repo] = GITHUB_REPOSITORY.split('/');
+
+const octokit = new Octokit({ auth });
+
+/**
+ * Gets the in progress workflow runs of a specified workflow.
+ *
+ * @param {string} workflow_id - Workflow id or file name
+ * @returns {Array} List of in progress workflow runs.
+ */
+const getInProgressWorkflowRuns = workflow_id => {
+  const params = {
+    owner,
+    repo,
+    workflow_id,
+    branch: 'main',
+    status: 'in_progress',
+  };
+
+  return octokit.rest.actions
+    .listWorkflowRuns(params)
+    .then(response => {
+      if (response.status !== 200) {
+        throw new Error(
+          `Response ${response.status} from ${response.url}. Aborting.`,
+        );
+      }
+      return response.data;
+    })
+    .then(({ workflow_runs }) => {
+      return workflow_runs;
+    });
+};
+
+/**
+ * Gets the commit hash of the last full deploy of an environment.
+ *
+ * @param {string} env - Name of environment
+ * @returns {string} Commit hash of the latest full deploy.
+ */
+const getLastFullDeployCommit = async env => {
+  const envBucketUrl = BUCKETS[env];
+  const buildTextUrl = new URL(path.join(envBucketUrl, 'BUILD.txt'));
+
+  const response = await fetch(buildTextUrl);
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch ${buildTextUrl}.\n${response.statusText}`);
+  }
+
+  const buildTextFile = await response.text();
+  const lines = buildTextFile.split('\n').filter(line => line);
```

Comment:
Filtering may not be needed if there aren't blank lines (or could do `.filter(Boolean)` as another approach)

Reply:
That's what I was thinking too. I don't see any blank lines in the files, but I noticed we [do this](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/b9dc75bfaf7dd6a0d6bf5b3356aea8cf8c9362df/packages/documentation/src/components/dashboard/DashboardDataFetch.jsx#L125-L127) for the FE support dashboard. Not sure if it's for a specific reason or just to be safe 🤔. `.filter(Boolean)` is definitely cleaner though, I'll update it to that